prov/efa/src/efa_cq.c

```diff
@@ -0,0 +1,535 @@
+/* SPDX-License-Identifier: BSD-2-Clause OR GPL-2.0-only */
+/* SPDX-FileCopyrightText: Copyright (c) 2013-2015 Intel Corporation, Inc.  All rights reserved. */
+/* SPDX-FileCopyrightText: Copyright Amazon.com, Inc. or its affiliates. All rights reserved. */
+
+#include <errno.h>
+#include <string.h>
+#include "config.h"
+#include <ofi_mem.h>
+#include "dgram/efa_dgram_ep.h"
+#include "efa.h"
+#include "efa_cq.h"
+#include "efa_av.h"
+#include <infiniband/verbs.h>
+
+
+static inline uint64_t efa_cq_opcode_to_fi_flags(enum ibv_wc_opcode opcode) {
+	switch (opcode) {
+	case IBV_WC_SEND:
+		return FI_SEND | FI_MSG;
+	case IBV_WC_RECV:
+		return FI_RECV | FI_MSG;
+	case IBV_WC_RDMA_WRITE:
+		return FI_RMA | FI_WRITE;
+	case IBV_WC_RECV_RDMA_WITH_IMM:
+		return FI_REMOTE_CQ_DATA | FI_RMA | FI_REMOTE_WRITE;
```

Comment:
If you have IMM for recv op code, do you need `FI_REMOTE_CQ_DATA` as well?

Reply:
I was referring to https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/rdm/efa_rdm_cq.c#L531

Reply:
Yeah, then for IBV_WC_RECV, when will you add FI_REMOTE_CQ_DATA?